### PR TITLE
Standardize presubmits for must-gather-operator

### PR DESCRIPTION
**Pull request by 2uasimojo**
Reorganize, rename, and rewrite the contents of the presumbit jobs for must-gather-operator according to the referenced Jira.

Jira: [OSD-4673](https://issues.redhat.com/browse/OSD-4673)

### CODE REVIEW

**Comment by 2uasimojo** [comment]
/hold

We first need to land https://github.com/openshift/boilerplate/pull/49 and pull it into [must-gather-operator](https://github.com/openshift/must-gather-operator).

**Comment by 2uasimojo** [comment]
@csheremeta @jharrington22 @dustman9000

**Comment by 2uasimojo** [comment]
`build` is failing because [this line](https://github.com/openshift/must-gather-operator/blob/e2ec68929c46718a93dd6c2b5dff59aca3b8ff9c/boilerplate/openshift/golang-osd-operator/standard.mk#L19):
```makefile
CONTAINER_ENGINE=$(shell command -v podman 2>/dev/null || command -v docker 2>/dev/null)
```
...is finding neither podman nor docker in the CI container. This is a place where it would be neat to use the image @arjunrn has been working on. However, it surprises me that docker doesn't exist in this container, so I'll debug further tomorrow.

`coverage` is failing because MGO still needs its [external codecov setup](https://github.com/openshift/ops-sop/blob/master/services/codecov.md#generate-the-codecovio-token) done. @dustman9000 or @csheremeta, is that something you could do?

File: ci-operator/config/openshift/must-gather-operator/openshift-must-gather-operator-master.yaml [hunk 1]
```diff
@@ -11,27 +11,31 @@ resources:
...
-- as: unit
+- as: test
   commands: |
-    set -x
-    go mod download
     make test
   container:
     from: src
+- as: coverage
+  commands: |
```
Comment (csheremeta) [inline ci-operator/config/openshift/must-gather-operator/openshift-must-gather-operator-master.yaml:30]:
```suggestion
  commands: |
    export CODECOV_TOKEN=$(cat /tmp/secret/CODECOV_TOKEN)
```
should we add this line?

**Comment by csheremeta** [comment]
codecov setup: #12345 12345

File: ci-operator/config/openshift/must-gather-operator/openshift-must-gather-operator-master.yaml (hunk 1 above)
Comment (2uasimojo) [reply ci-operator/config/openshift/must-gather-operator/openshift-must-gather-operator-master.yaml:30]: Apparently so :)

**Comment by 2uasimojo** [comment]
Okay, podman and docker are both legitimately absent from the container.

**Comment by 2uasimojo** [comment]
So apparently I can't run docker inside a container. Who knew?

Since `make build` was just running `docker build`, I've cut over to using the `images:` config instead. This looks promising as a strategy -- but it turns out MGO's docker build is busted anyway. So we'll need to fix that before we can move this forward.

**Comment by 2uasimojo** [comment]
We'll use #12961 instead.

**Pull request closed without merging**